### ISSUE TITLE
[Feature request] zod async refinement

### ISSUE THREAD

**Issue by paulsouche** [ts-rest/ts-rest#734]
**Is your feature request related to a problem? Please describe.**
ts-rest does not allow me to do async refinement on query with zod. Error

```
Async refinement encountered during synchronous parse operation. Use .parseAsync instead.
```

**Describe the solution you'd like**
The async refinement should be possible

**Describe alternatives you've considered**
Do the validation in the handler (workaround) / change library called during refinement (impossible)

**Additional context**
My use case is to validate a JWT with jsonwebtoken library

**Comment by oliverbutler** [comment]
Given we're working on #743 in which async validation doesn't seem key to the standard 

> Avoid returning Promise from ~standard.validate() unless absolutely necessary. Some third-party libraries may not support async validation.

Im going to close as "wont do", this may be done as part of that PR but equally if not in the standard we won't likely add it!

**Comment by paulsouche** [comment]
@oliverbutler Sure ! No problem ! Looking forward for this new standard and hopefully `async` validation would be a thing at some point in this `async` by design language. The reason

> Some third-party libraries may not support `async` validation.

Really looks fragile for me. `await` a thing that is `async` or not should not be a problem.

Anyway, thank you for the great job !

**Comment by oliverbutler** [comment]
You're right it'll likely be part of it, I haven't gotten deep into it, but this standard is definitely going to be a huge improvement for us.

**Comment by aidant** [comment]
I prototyped supporting async validation in https://github.com/ts-rest/ts-rest/pull/751 but ran into some issues since not all places that perform validation are async. We could conditionally perform async validation where its supported, but I believe that will lead to undesirable behaviour where x feature is only supported in y scenarios. Alternatively it might be possible to convert the sync functions to async to support this feature, but I considered that out of scope for the standard schema PR.
